core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -61,7 +60,9 @@ public String getName()
             WarningCollector warningCollector)
     {
         Session session = stateMachine.getSession();
-        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getTableName());
+        QualifiedObjectName originalTableName = createQualifiedObjectName(session, statement, statement.getTableName());
+        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
+        QualifiedObjectName tableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(originalTableName);
```

Comment:
redirections for views and MVs aren't handled, the API is only for tables. At other similar places, redirected name is only used after view/mview checks are done. I think we should do the same here. (for current usage it doesn't matter much since connector wouldn't redirect for a view anyway)

Reply:
Good point. We should add redirects to views and MVs as well.
Want file an issue?

Reply:
https://github.com/trinodb/trino/issues/11225